jdaviz/app.py

```diff
@@ -1816,18 +1824,37 @@ def _reparent_subsets(self, old_parent, new_parent=None):
                                     dummy_x = old_xc + r
                                     x2, y2 = pixel_to_pixel(old_parent.coords, new_parent.coords,
                                                             dummy_x, old_yc)
-                                    new_radius = np.abs(x2 - x)
+                                    # Need to use x and y in this radius calculation because the
+                                    # new orientation is likely rotated compared to the original.
+                                    new_radius = np.sqrt((x2 - x)**2 + (y2 - y)**2)
                                     setattr(roi, att, new_radius)
 
                         elif type(roi) is RectangularROI:
-                            x_min, y_min = pixel_to_pixel(old_parent.coords, new_parent.coords,
-                                                          roi.xmin, roi.ymin)
-                            x_max, y_max = pixel_to_pixel(old_parent.coords, new_parent.coords,
-                                                          roi.xmax, roi.ymax)
-                            roi.xmin = x_min
-                            roi.xmax = x_max
-                            roi.ymin = y_min
-                            roi.ymax = y_max
+                            x1, y1 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmin, roi.ymin)
+                            x2, y2 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmin, roi.ymax)
+                            x3, y3 = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                    roi.xmax, roi.ymin)
+
+                            # Calculate new width and height from possibly rotated result
+                            new_width = np.sqrt((x3-x1)**2 + (y3-y1)**2)
+                            new_height = np.sqrt((x2-x1)**2 + (y2-y1)**2)
+
+                            # Convert center
+                            new_center = pixel_to_pixel(old_parent.coords, new_parent.coords,
+                                                        old_xc, old_yc)
+
+                            # New min/max before applying theta
+                            roi.xmin = new_center[0] - new_width/2
+                            roi.xmax = new_center[0] + new_width/2
+                            roi.ymin = new_center[1] - new_height/2
+                            roi.ymax = new_center[1] + new_height/2
+
+                        # Account for rotation between orientations
+                        if hasattr(roi, "theta"):
+                            angle = getattr(roi, "theta")
+                            setattr(roi, "theta", angle - np.deg2rad(relative_angle))
```

Comment:
You already check for `hasattr` above, can't you just access it directly like this? Am I missing something?
```suggestion
                            roi.theta = roi.theta - np.deg2rad(relative_angle)
```

Reply:
Did this locally and added you as a co-author.